wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/json/JacksonPreStatement.java

```diff
@@ -33,100 +30,53 @@
 import java.util.*;
 
 /**
- * Helper class storing a statement as represented in JSON. This
- * is a Statement missing a subject.
+ * Helper class for deserializing statements from JSON.
  * 
  * @author Antonin Delpeuch
+ * @author Thomas Pellissier Tanon
  *
  */
-@JsonInclude(Include.NON_EMPTY)
-@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonIgnoreProperties(ignoreUnknown=true)
 public class JacksonPreStatement {
 
-	/**
-	 * Id of this statement.
-	 *
-	 * @see Statement#getStatementId()
-	 */
-	private final String id;
+	private final String statementId;
 
-	/**
-	 * Rank of this statement.
-	 */
-	@JsonSerialize(using = StatementRankSerializer.class)
-	@JsonDeserialize(using = StatementRankDeserializer.class)
 	private final StatementRank rank;
 
-	@JsonDeserialize(contentAs=ReferenceImpl.class)
 	private final List<Reference> references;
 
-	/**
-	 * The main snak of this statement.
-	 */
-	@JsonDeserialize(contentAs=SnakImpl.class)
-	private final Snak mainsnak;
+	private final Snak mainSnak;
 
-	/**
-	 * A map from property id strings to snaks that encodes the qualifiers.
-	 * We use the explicit type {@link SnakImpl} because Jackson does
-	 * not know yet how to specify subclasses for interfaces nested in two
-	 * containers.
-	 */
-	@JsonIgnore
 	private final Map<String, List<Snak>> qualifiers;
 	
-	/**
-	 * List of property string ids that encodes the desired order of qualifiers,
-	 * which is not specified by the map.
-	 */
-	private final List<String> propertyOrder;
+	private final List<String> qualifiersOrder;
 
-	/**
-	 * Constructor.
-	 */
-	public JacksonPreStatement(
-			String id,
+	private JacksonPreStatement(
+			String statementId,
 			StatementRank rank,
 			Snak mainsnak,
 			Map<String, List<Snak>> qualifiers,
-			List<String> propertyOrder,
+			List<String> qualifiersOrder,
 			List<Reference> references) {
-		if (id == null) {
-			id = "";
-		}
-		this.id = id;
-		Validate.notNull(rank, "No rank provided to create a statement.");
+		this.statementId = statementId;
 		this.rank = rank;
-		Validate.notNull(mainsnak, "No main snak provided to create a statement.");
-		this.mainsnak = mainsnak;
-		if (qualifiers != null) {
-			this.qualifiers = qualifiers;
-		} else {
-			this.qualifiers = Collections.emptyMap();
-		}
-		if (propertyOrder != null) {
-			this.propertyOrder = propertyOrder;
-		} else {
-			this.propertyOrder = Collections.emptyList();
-		}
-		if (references != null) {
-			this.references = references;
-		} else {
-			this.references = Collections.emptyList();
-		}
```

Comment:
Are we still be able to deserialize statements where references or qualifiers are not provided? (Maybe nobody produces JSON like that actually?)

Reply:
Yes, we are still able to deserialize these statements (See the implementation.json.TestStatement tests). It does not cost us much to still support this case and there is a meaningful fallback (no references/qualifiers)

Reply:
okay so Jackson automatically passes empty containers for properties that are not supplied?

Reply:
It passes null for the not supplied properties. It's why we need to check for null.

Reply:
That's what I thought - so where are these checks gone?

Reply:
In the StatementImpl constructor. No need to have them twice.

Reply:
sure, thanks